### DISCUSSION TITLE
MQTT Nachkommastellen Yieldtotal

### DISCUSSION THREAD

**Discussion by m0tschi** [tbnobody/OpenDTU#702] [Q&A]
Moin zusammen,
Ist es im Code einfach möglich die vollen Nachkommastellen übermitteln zu lassen? In der WebApp sieht man 123,456 kw/h in Mqtt jedoch wird nur 123,45 übertragen.
Oder wäre das von Aufwand her eher ein Feature Wunsch?

Danke schonmal für die Hilfe! :)

**Comment by Solarteur** [comment]
In der WebUI werden unter "Total Yield Total" auch "nur" zwei Nachkommastellen angezeigt - und das ist aus meiner Sicht gut so. Jeder kann selbst nachrechnen, wie hoch bei einem Messfehler von 1% (und das ist sehr niedrig angesetzt, realistisch sind eher 2% bis 3%) die Abweichung des gemessenen Wertes vom wirklichen Wert sein kann. 
Dazu kommt, dass unter "Total Yield Total" die **DC**-Werte aufsummiert werden - für den **AC**-Ertrag ist dann noch der Wirkungsgrad des WR zu berücksichtigen, auch keine Konstante.

**Reply by m0tschi** [reply]
Da stimme ich dir Teilweise zu, aber da ich meine Auswertung gerne über den YieldTotal des WR machen möchte und ich mehrere WR habe welche auf unterschiedliche Zähler laufen (an einer DTU) wäre es am einfachsten wenn man die Wh anhand der Zählerstände errechnet.
Bei der Einzelansicht der WR sieht man die 3 Nachkommastellen, zumindest in der aktuellen Version.

**Comment by tbnobody** [comment]
Ich habe gerade gesehen, dass die String Klasse einen Default Wert von 2 Nachkommastellen verwendet. Das ist in der Tat so nicht beabsichtigt. Es sollten alle gültigen Nachkommastellen wie im Inverter verfügbar auch ausgegeben werden. Wird in einem der nächsten Commits behoben.
